pkg/registry/garden/backupinfrastructure/strategy.go

```diff
@@ -82,6 +87,12 @@ func mustIncreaseGeneration(oldBackupInfrastructure, newBackupInfrastructure *ga
 		return true
 	}
 
+	oldPurpose = oldBackupInfrastructure.ObjectMeta.Annotations[gardencorev1alpha1.GardenPurpose]
```

Comment:
Also, handle the purpose annotation here: https://github.com/gardener/gardener/blob/master/pkg/registry/garden/shoot/strategy.go#L106.

Reply:
Done.